addons/mrp/views/mrp_production_views.xml

```diff
@@ -90,7 +91,7 @@
                                 <field name="product_uom_id" options="{'no_open':True,'no_create':True}" groups="product.group_uom" attrs="{'readonly': [('has_moves', '=', True)]}"/>
                                 <button type="action"
                                     name="%(mrp.action_change_production_qty)d"
-                                    string="Update" states="confirmed,planned,progress" class="oe_link" attrs="{'invisible': [('state','=','done')]}"/>
+                                    string="Update" states="planned,progress" class="oe_link" attrs="{'invisible': ['|', ('state', '=', 'done'), ('id', '=', False)]}"/>
```

Comment:
If we want to show or hide "update link" based on some condition, condition will not work for provided states (in the field attribute), so we need to remove particular state ("confirmed") from the states (field attribute)